io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -637,11 +637,19 @@ impl LogicalVolume for Lvol {
         unsafe { spdk_blob_is_read_only(self.blob_checked()) }
     }
 
-    /// Return the size of the Snapshot in bytes.
+    /// Return the size of the Logical Volume in bytes.
     fn size(&self) -> u64 {
         self.as_bdev().size_in_bytes()
     }
 
+    /// Return the committed size of the Logical Volume in bytes.
+    fn committed(&self) -> u64 {
+        match self.is_read_only() {
```

Comment:
Is this ok or should this be is_snapshot?

Reply:
If either of `data` or `metadata` is readonly in blob, We are marking read_only flag as true.
Now from SPDK not finding any instance where `metadata` is true but `data` is false. If such situation comes in future, then possibly `is_read_only` is not a better option. IMHO `is_snapshot` is safe for our scenario.

Reply:
Even if metadata is ro the snapshot can still allocate more clusters?
I've changed to is_snapshot as it seems safer then.

Reply:
From SPDK documentation, metadata section is fixed during blob initialization. In theory, I think We can add cluster if metadata is ro. But need to check once in the code.

Reply:
Then I think let's merge this as is for now, we can improve later with a more general read_only, if possible at all.